pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -208,6 +208,33 @@ func (e *endpointsController) buildIstioEndpoints(endpoint interface{}, host hos
 	return endpoints
 }
 
+func (e *endpointsController) buildIstioEndpointsWithService(name, namespace string, host host.Name) []*model.IstioEndpoint {
+
+	ep, err := listerv1.NewEndpointsLister(e.informer.GetIndexer()).Endpoints(namespace).Get(name)
+	if err != nil || ep == nil {
+		log.Debugf("Get endpoint %s/%s failed %v", name, namespace, err)
+		return nil
+	}
+
+	endpoints := make([]*model.IstioEndpoint, 0)
+	for _, ss := range ep.Subsets {
```

Comment:
The code below is duplicate with ` buildIstioEndpoints`, will abstract it if this is the right direction.